open-grant-proposals/open-proposal-blockchain-package-manager.md

```diff
@@ -0,0 +1,183 @@
+* FileCoin added to the Blockdaemon marketplace
+* (Optional) monitoring support via Blockdaemon monitoring services
+
+## Development Roadmap
+
+### Milestone 1
+
+* Source-code for a basic bpm plugin to launch a FileCoin node
+* Basic bpm plugin deployed to BPM repository
+* Documentation on how to launch a FileCoin node using bpm
+
+This milestone will take 4 weeks
+
+People working on this milestone
+* 1 x QA
+* 1 x Software Engineers
+* 1 x Project Manager
+
+Funding required: €15.000
+
+Acceptance criteria
+
+* The user is provided BPM cli that will download the BPM FileCoin package  that will start a FileCoin node
+* The user can install on Ubuntu or OSX running on GCE, AWS, and Azure
+* The user can run the individual BPM commands:
+    * list - Lists all available version of the FileCoin package
+    * install - Installs the latest version of the FileCoin package
+    * configure - Creates the base configuration for a FileCoin node
+    * start - Creates all necessary secrets and starts a FileCoin node
+    * stop - Stops the FileCoin node (and optionally purges all data)
+    * status - Returns the status of all configured nodes on the system
+    * help - Help about any command
+
+### Milestone 2
+
+* Monitoring agents will be added to the FileCoin package
+* Advanced FileCoin configuration options will be added
+
+This milestone will take 4 weeks
+
+People working on this milestone
+* 1 x QA
+* 2 x Software Engineers
+* 1 x Project Manager
+
+Funding required: €15.000
+
+Acceptance criteria:
+
+* Logs, telemetry and node status will be sent down the Blockdaemon data pipeline
+* User can query for node state with the Blockdaemon node API
+* Advanced configuration options for the FileCoin node are supported via the `configure` command
+
+### Milestone 3
+
+* Automated Test Suite
+
+This milestone will take 4 weeks
+
+People working on this milestone
+* 2 x QA
+* 1 x Software Engineers
+* 1 x Project Manager
+
+Funding required: €10.000
+
+Acceptance criteria:
+
+* User should be able to run the automated test suite and all tests should pass
+* The user runs/installs the node with BPM and can run “bpm test filecoin”, which will run a test suite against the running setup.
+
+### Milestone 4
+
+* Add FileCoin to the Blockdaemon Marketplace for managed nodes with a dedicated marketing page
+* Blockdaemon will analyze the metrics, telemetry data and logs for anomalies
+* Blockdaemon will launch and maintain one FileCoin node
+
+This milestone will take 4 weeks
+
+People working on this milestone
+* 1 x System administrator
+* 1 x Data engineer
+* 1 x Web developer
+* 1 x Project Manager
+
+Funding required: €10.000
+
+Acceptance criteria:
+
+* Complete Product
+* First users start onboarding
+* Blockdaemon hosted FileCoin node accessible
+
+## Total Budget Requested
+
+€50.000
+
+## Maintenance and Upgrade Plans
+
+The team will maintain the package long-term and release new package versions with every new FileCoin release.
```

Comment:
Is the plan for your team to maintain this on your own (i.e. without further grant support)?